### PR TITLE
customize repo to accomodate cmm kafka env

### PR DESCRIPTION
**Pull request by deanmax**
Repo customization to fit CMM environment:

- update README
- update kafka / scala version to align with cmm kafka version
- update producer/consumer setting to get it work in cmm kafka env
- rename build output file for better distinguishing with other kafka monitors to be created

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -51,16 +51,17 @@ mvn:
 	@echo ===================================
 	make lib-mvn
 	./mvnw clean package
-	ln -sfv target/check_kafka-*.jar check_kafka.jar
+	ln -sfv target/check_kafka-*.jar check_kafka_broker.jar
```
Comment (bilsch) [inline Makefile:54]: is this just making the symlink more specific?

File: README.md [hunk 2]
```diff
@@ -1,30 +1,7 @@
-Kafka Scala API Tester - Advanced Nagios Plugin / CLI Tool with Kerberos support
+Kafka Scala API Tester - Advanced Nagios Plugin / CLI Tool with Kerberos support (CMM customized)
```
Comment (bilsch) [inline README.md:1]: if we want to get the change merged in I would back this change out

Comment (bilsch) [reply README.md:1]: are all of the changes to readme.md still valid?

File: conf/kafka_cli_jaas.conf [hunk 3]
```diff
@@ -13,10 +13,7 @@
 //
 
 KafkaClient {
-    com.sun.security.auth.module.Krb5LoginModule required
```
Comment (bilsch) [inline conf/kafka_cli_jaas.conf]: this feels odd to me. I get why the change but there should be jaas configurations for both kerberos and plaintext ( or others ). This is changing the default which probably is not the right way to go.

This comment applies to the other jaas configs as well

File: pom.xml [hunk 4]
```diff
@@ -45,14 +45,14 @@
       Kafka 0.10 API bug:
       Cannot auto-commit offsets for group ... since the coordinator is unknown
     -->
-    <kafka.version>0.9.0.1</kafka.version>
+    <kafka.version>0.11.0.1</kafka.version>
```
Comment (bilsch) [inline pom.xml:48]: I suspect you may get some pushback here as the original author was doing work with hdp. Afaik there is no hdp release which includes 0.11. Nothing to change now just something to keep in mind

File: src/main/resources/consumer.properties [hunk 5]
```diff
@@ -15,7 +15,6 @@
 # Intentionally not respecting 'group.id' setting here as I enforce a unique group id to avoid any potential group sharing whch might inadvertantly cause a message being routed to another consumer. This way I guarantee we should receive back our unique generated message as part of the test
 
 bootstrap.servers = localhost:9092
-security.protocol = PLAINTEXT
```
Comment (bilsch) [inline src/main/resources/consumer.properties:18]: ditto with the jaas config changes I suspect there will need to be different properties configs for kerberos and non-kerberos

**Review by bilsch — COMMENTED** [review]
I suspect there will need to be some changes if this change is going to be sent up to the original author for merge to the main code base

File: Makefile (hunk 1 above)
Comment (deanmax) [reply Makefile:54]: yes, I just thought there would be more kafka checks in the future. check_kafka is too vague and might mislead ppl among all other checks.

File: README.md (hunk 2 above)
Comment (deanmax) [reply README.md:1]: I'm not considering merging our fork back to upstream. The changs here are all serving cmm purpose.

**Comment by deanmax** [comment]
The fork of this repo is for cmm internal need at this mement, so no need to concern merging back to the upstream repo. As for the kereros vs. non-kerberos config, we can change it later once our kafka puppet manifest has started using kerberos. At this moment, this check works in our kafka env.

File: README.md (hunk 2 above)
Comment (bilsch) [reply README.md:1]: well, my use of kafka has kerberos as a requirement so ;)

**Comment by deanmax** [comment]
Kerberos support is not yet finished at this moment. This preliminary PR is to facilitate kafka nagios plugin RPM building. Will keep working on kerberos support later with a different PR.
